libminifi/include/core/PropertyValidation.h

```diff
@@ -308,6 +288,66 @@ class TimePeriodValidator : public PropertyValidator {
   }
 };
 
+class RegexValidator : public PropertyValidator {
+ public:
+  explicit RegexValidator(const std::string& pattern)
+  : PropertyValidator("REGULAR_EXPRESSION_VALIDATOR"){
+    regex_ = utils::Regex(pattern);
+  }
+  ValidationResult validate(const std::string &subject, const std::shared_ptr<minifi::state::response::Value> &input) const {
+    return validate(subject, input->getStringValue());
+  }
+
+  ValidationResult validate(const std::string &subject, const std::string &input) const {
+    bool result = regex_.match(input);
+    return ValidationResult::Builder::createBuilder().withSubject(subject).withInput(input).isValid(result).build();
+  }
+ private:
+  mutable utils::Regex regex_;
+};
+
+template <typename T, typename std::enable_if<std::is_base_of<PropertyValidator, T>::value>::type* = nullptr>
+class ListValidator : public PropertyValidator{
+ public:
+  explicit ListValidator(T val)
+  : PropertyValidator("LIST_VALIDATOR"),
```

Comment:
The name should somehow contain the underlying validator's name to work properly with C2. 
This should be adjusted to the way Java MiNiFi or NiFi works.